src/main/scala/com/wavesplatform/matcher/api/MatcherApiRoute.scala

```diff
@@ -234,14 +235,8 @@ case class MatcherApiRoute(wallet: Wallet,
       )
     ))
   def historyDelete: Route = (path("orderbook" / AssetPairPM / "delete") & post) { p =>
-    withAssetPair(p) { pair =>
-      json[CancelOrderRequest] { req =>
-        if (req.isSignatureValid()) {
-          log.trace(s"Deleting ${req.orderId.fold(s"all orders for ${req.sender}")(_.base58)}")
-          (orderHistory ? DeleteOrderFromHistory(pair, req.sender, req.orderId, NTP.correctedTime()))
-            .mapTo[MatcherResponse]
-        } else InvalidSignature
-      }
+    json[CancelOrderRequest] { req =>
+      req.orderId.fold[MatcherResponse](NotImplemented("Batch order deletion is not supported yet"))(OrderDeleted)
```

Comment:
just remove batch deletion support

Reply:
I can't, CancelOrderRequest is used in other methods :) A current implementation is ok, I think...